src/types/arbitrage/skipLoop.ts

```diff
@@ -95,16 +86,23 @@ export class SkipLoop extends MempoolLoop {
 	/**
 	 *
 	 */
-	private async skipTrade(arbTrade: { path: Path; offerAsset: Asset; profit: number }, toArbTrade: MempoolTrade) {
+	private async skipTrade(arbTrade: OptimalTrade, toArbTrade: MempoolTrade) {
+		if (
+			!this.botConfig.useSkip ||
+			this.botConfig.skipRpcUrl === undefined ||
+			this.botConfig.skipBidRate === undefined ||
+			this.botConfig.skipBidWallet === undefined
+		) {
+			console.error("please setup skip variables in the config environment file", 1);
+			return;
+		}
 		const bidMsg: MsgSend = MsgSend.fromJSON({
 			fromAddress: this.account.address,
-			toAddress: this.skipBidWallet,
+			toAddress: this.botConfig.skipBidWallet,
 			amount: [
 				{
-					denom: isNativeAsset(this.offerAssetInfo)
-						? this.offerAssetInfo.native_token.denom
-						: this.offerAssetInfo.token.contract_addr,
-					amount: String(Math.max(Math.round(arbTrade.profit * this.skipBidRate), 651)),
+					denom: this.botConfig.offerAssetInfo,
+					amount: String(Math.max(Math.round(arbTrade.profit * this.botConfig.skipBidRate), 651)),
```

Comment:
Gave this a look after someone in the discord brought up an issue. This feels potentially wrong since in the old version it pulls the denom directly whereas after the change it places the entire asset info into the denom field - and from the looks of it the asset info object is `{ native_token: { denom: envs.BASE_DENOM } }`